File: models/Exame.php
<?php

require_once 'Paciente.php';

class Exame {
    private $idExame;
    private $paciente; // Objeto Paciente
    private $laboratorio;
    private $exameTexto;
    private $dataExame;
    private $resultado;

    public function __construct(Paciente $paciente, $laboratorio, $exameTexto, $idExame = null, $dataExame = null, $resultado = null) {
        $this->paciente = $paciente;
        $this->laboratorio = $laboratorio;
        $this->exameTexto = $exameTexto;
        $this->idExame = $idExame;
    }

    // Getters e Setters
    public function getIdExame() {
        return $this->idExame;
    }

    public function setIdExame($idExame) {
        $this->idExame = $idExame;
    }

    public function getPaciente() {
        return $this->paciente;
    }

    public function setPaciente(Paciente $paciente) {
        $this->paciente = $paciente;
    }

    public function getLaboratorio() {
        return $this->laboratorio;
    }

    public function setLaboratorio($laboratorio) {
        $this->laboratorio = $laboratorio;
    }

    public function getExameTexto() {
        return $this->exameTexto;
    }

    public function setExameTexto($exameTexto) {
        $this->exameTexto = $exameTexto;
    }

    public function getDataExame() {
        return $this->dataExame;
    }

    public function setDataExame($dataExame) {
        $this->dataExame = $dataExame;
    }

    public function getResultado() {
        return $this->resultado;
    }

    public function setResultado($resultado) {
        $this->resultado = $resultado;
    }

    // Método salvar usando conexão (pode ser usado ou substituído pelo DAO)
    public function salvar($conn) {
        $sql = "INSERT INTO exames (idPaciente, laboratorio, exameTexto) VALUES (?, ?, ?)";

        $stmt = $conn->prepare($sql);

        if (!$stmt) {
            throw new Exception("Erro na preparação da query: " . $conn->error);
        }

        $idPaciente = $this->paciente->getIdPaciente();

        $stmt->bind_param("sss", $idPaciente, $this->laboratorio, $this->exameTexto);

        if (!$stmt->execute()) {
            throw new Exception("Erro ao salvar exame: " . $stmt->error);
        }

        $this->idExame = $stmt->insert_id; // Pega o ID gerado no banco

        $stmt->close();
    }
}

File: models/java/Exame.java
package lab;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class Exame {
    private Paciente paciente;
    private String laboratorio;
    private String exameTexto;

    public Exame(Paciente paciente, String laboratorio, String exameTexto) {
        this.paciente = paciente;
        this.laboratorio = laboratorio;
        this.exameTexto = exameTexto;
    }

    public void salvar(Connection conn) throws SQLException {
        String sql = "INSERT INTO exames (idPaciente, laboratorio, exameTexto) VALUES (?, ?, ?)";

        try (PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, paciente.getId());
            stmt.setString(2, laboratorio);
            stmt.setString(3, exameTexto);

            stmt.executeUpdate();
        } catch (SQLException e) {
            throw new SQLException("Erro ao salvar exame: " + e.getMessage(), e);
        }
    }

    public static void listar(Connection conn) {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'listar'");
    }
}
